GSoC25/NEF/Emeddings.py

```
@@ -0,0 +1,171 @@
+# Extract unique predicate rows by URI (keep first label/comment seen)
+df["predicate"] = df["predicate"].astype(str)
+uniq = df.drop_duplicates(subset=["predicate"], keep="first").reset_index(drop=True)
+
+predicates = uniq["predicate"].tolist()
+labels = uniq["label"].astype(str).tolist()
+comments = uniq["comment"].apply(lambda x: str(x) if not pd.isna(x) else "").tolist()
+
+print(f"Unique predicate URIs: {len(predicates)}")
+
+# === Build texts to embed (semantic, not raw URIs) ===
+texts = [make_label_text(u, l, c if c else None) for u, l, c in zip(predicates, labels, comments)]
```

Ruff (0.14.2)

Ambiguous variable name: `l`

(E741)